src/stores/AllEventsStore.test.js

```diff
@@ -0,0 +1,91 @@
+import { Routes } from 'constants';
+jest.mock('../network/graphql/queries'); // block and manually mock our backend
+import AllEventsStore from './AllEventsStore'; // eslint-disable-line
+
+/** mock necessary http needed modules */
+const localStorageMock = {
+  getItem: jest.fn(),
+  setItem: jest.fn(),
+  clear: jest.fn(),
+};
+const navigatorMock = {};
+global.navigator = navigatorMock;
+global.localStorage = localStorageMock;
+
+/** test */
+describe('AllEventsStore', () => {
+  let store;
+  const app = {
+    sortBy: 'ASC',
+    wallet: { addresses: 0 },
+    global: { syncBlockNum: 0 },
+    refreshing: false,
+    ui: { location: 0 },
+  }; // mock the appstore
+  beforeEach(() => {
+    store = new AllEventsStore(app); // create a new instance before each test case
+  });
+
+  /** all following test cases target specifically to the mock backend, eg. network/graphql/__mocks__/queries.js */
+  it('Init', async () => {
+    const curListNum = store.list.length;
+    expect(curListNum).toBe(0);
+    expect.assertions(4);
+    await store.init();
+    expect(app.ui.location).toBe(Routes.ALL_EVENTS); // test webpage location
+    expect(store.list.length).toBe(curListNum + store.limit); // test if list got new cards
+    expect(store.loading).toBe(false); // test runInAction changes
```

Comment:
can remove unnecessary comments. these are self-explanatory by the `expect.toBe` as they read like sentences

Reply:
removed